# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1786750962 -7200
# Parent  10e2954a9afd11d392c1c7e2315b94d2c6727ce8
Feature: Linux systray: Tweaks for compiling.

The Mozilla build pins GLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_42
against a sysroot whose glib predates g_autofree (added in 2.44).

diff --git a/third_party/appindicator/ayatana-appindicator.c b/third_party/appindicator/ayatana-appindicator.c
--- a/third_party/appindicator/ayatana-appindicator.c
+++ b/third_party/appindicator/ayatana-appindicator.c
@@ -17,8 +17,10 @@
  */
 
 #define _GNU_SOURCE
 
+#include <stdlib.h>
+
 #include "ayatana-appindicator.h"
 #include "ayatana-appindicator-enum-types.h"
 #include "application-service-marshal.h"
 #include "dbus-shared.h"
@@ -1526,14 +1528,15 @@ static gchar* appendSnapPrefix (const gc
     const gchar *sSnap = getSnapPrefix ();
 
     if (sSnap && path)
     {
-        g_autofree gchar *sCanonPath = realpath (path, NULL);
+        gchar *sCanonPath = realpath (path, NULL);
         gboolean bPrefix = g_str_has_prefix (sCanonPath, "/tmp/");
 
         if (bPrefix)
         {
             g_warning ("Using '/tmp' paths in SNAP environment will lead to unreadable resources");
+            g_free (sCanonPath);
 
             return NULL;
         }
 
@@ -1550,9 +1553,9 @@ static gchar* appendSnapPrefix (const gc
         gboolean bRuntimePrefix = g_str_has_prefix (sCanonPath, sRuntime);
 
         if (bSnapPrefix || bHomePrefix || bCachePrefix || bConfigPrefix || bDataPrefix || bRuntimePrefix)
         {
-            return g_strdup (sCanonPath);
+            return sCanonPath;
         }
 
         for (gint nDir = 0; nDir < G_USER_N_DIRECTORIES; ++nDir)
         {
@@ -1560,13 +1563,14 @@ static gchar* appendSnapPrefix (const gc
             gboolean bPrefix = g_str_has_prefix (sCanonPath, sDir);
 
             if (bPrefix)
             {
-                return g_strdup (sCanonPath);
+                return sCanonPath;
             }
         }
 
         gchar *sPath = g_build_path (G_DIR_SEPARATOR_S, sSnap, sCanonPath, NULL);
+        g_free (sCanonPath);
 
         return sPath;
     }
 
